src/template.rs

```diff
@@ -141,13 +164,86 @@ pub mod django_rusty_templates {
     }
 
     impl Engine {
-        #[allow(dead_code)]
+        fn get_template_loaders<'py>(
+            py: Python<'py>,
+            template_loaders: Bound<'_, PyIterator>,
+        ) -> PyResult<Vec<Loader>> {
+            template_loaders
+                .map(|template_loader| template_loader.and_then(|template_loader| Self::find_template_loader(py, template_loader)))
+                .collect()
+        }
+
         fn find_template_loader<'py>(
-            _py: Python<'py>,
-            _loader: &str,
-            _args: Option<Bound<'py, PyAny>>,
-        ) -> PyResult<Bound<'py, PyAny>> {
-            todo!()
+            py: Python<'py>,
+            loader: Bound<'_, PyAny>,
+        ) -> PyResult<Loader> {
+            if let Ok(loader_str) = loader.downcast::<PyString>() {
+                return Self::map_loader(py, loader_str.to_str()?, None);
+            }
+
+            let (loader_path, args) = unpack(loader.clone()).map_err(|e| {
+                ImproperlyConfigured::new_err(format!(
+                    "Invalid template loader: {loader}. {}",
+                    e.value(py),
+                ))
+            })?;
+
+            Self::map_loader(py, loader_path.to_str()?, Some(args))
+        }
+
+        fn map_loader(
+            py: Python<'_>,
+            loader_path: &str,
+            args: Option<Bound<'_, PyAny>>,
+        ) -> PyResult<Loader> {
```

Comment:
```suggestion
        fn map_loader(
            &self,
            py: Python<'_>,
            loader_path: &str,
            args: Option<Bound<'_, PyAny>>,
        ) -> PyResult<Loader> {
```

Reply:
This issue is that `get_template_loaders` is called in the `fn new` implementation, and at this point, we don't have `self` constructed. i think I have to drill down encoding in `get_template_loaders` -> `find_template_loader` - `map_loader`